### PR TITLE
Rubocop: test/*

### PR DESCRIPTION
**Pull request by pathawks**
#4885

### CODE REVIEW

File: test/test_configuration.rb [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
       @mixed_keys = Configuration[{
-        'markdown' => 'kramdown',
-        :permalink => 'date',
-        'baseurl'  => '/',
-        :include   => ['.htaccess'],
-        :source    => './'
+        "markdown" => "kramdown",
+        :permalink => "date",
+        "baseurl"  => "/",
+        :include   => [".htaccess"],
```
Comment (envygeeks) [inline test/test_configuration.rb]: This should be %w(.htaccess), did it not throw an error for that?

File: test/test_configuration.rb [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
       @string_keys = Configuration[{
-        'markdown'  => 'kramdown',
-        'permalink' => 'date',
-        'baseurl'   => '/',
-        'include'   => ['.htaccess'],
-        'source'    => './'
+        "markdown"  => "kramdown",
+        "permalink" => "date",
+        "baseurl"   => "/",
+        "include"   => [".htaccess"],
```
Comment (envygeeks) [inline test/test_configuration.rb]: ""

File: test/test_configuration.rb (hunk 1 above)
Comment (pathawks) [inline test/test_configuration.rb]: [No sir](https://travis-ci.org/jekyll/jekyll/jobs/132733086#L391), it did not.

File: test/test_configuration.rb [hunk 3]
```diff
@@ -140,108 +155,141 @@ class TestConfiguration < JekyllUnitTest
...
-      allow(SafeYAML).to receive(:load_file).with(@paths[:other]).and_return({"baseurl" => "http://wahoo.dev"})
+      allow(SafeYAML).to receive(:load_file).with(@paths[:other]).and_return(
+        { "baseurl" => "http://wahoo.dev" }
+      )
       allow($stdout).to receive(:puts).with("Configuration file: #{@paths[:other]}")
-      assert_equal Utils.deep_merge_hashes(@@defaults, { "baseurl" => "http://wahoo.dev" }), Jekyll.configuration({ "config" => @paths[:other] })
+      assert_equal(
+        Utils.deep_merge_hashes(
+          defaults,
+          { "baseurl" => "http://wahoo.dev" }
```
Comment (envygeeks) [inline test/test_configuration.rb]:
```ruby
Utils.deep_merge(arg1, {
  :key => :val
})
```

**Comment by pathawks** [comment]
✅ Rubocop Approves

File: test/test_configuration.rb [hunk 4]
```diff
@@ -1,23 +1,25 @@
-require 'helper'
+# rubocop:disable Metrics/ClassLength
```
Comment (pathawks) [inline test/test_configuration.rb]: I wonder if we should just disable `Metrics/ClassLength` for all of `test/`?

**Comment by envygeeks** [comment]
:shipit:

File: test/test_configuration.rb (hunk 4 above)
Comment (envygeeks) [inline test/test_configuration.rb]: The only way to do that would be to exclude all of tests from the metrics (I'm not against this, I wonder what @parkr thinks?)  Other than that there is no way other than by file.

Comment (pathawks) [inline test/test_configuration.rb]: I believe we can disable just this metric from `.rubocop.yml`

If you don’t mind, I’ll give it a shot

Comment (envygeeks) [inline test/test_configuration.rb]: Yeah, I say go for it.

Comment (parkr) [inline test/test_configuration.rb]: @pathawks Yep, you can safely exclude all of `test/` from `Metrics/ClassLength`. I still would like them tested for the other cops as they're arguably the worst part of our codebase 😆

**Comment by pathawks** [comment]
These merge conflicts are *killing me* 😭

**Comment by envygeeks** [comment]
haha, yeah that'll happen when you send a million pull requests :P

File: .rubocop.yml [hunk 5]
```diff
@@ -68,6 +57,7 @@ Metrics/ClassLength:
   Max: 240
   Exclude:
   - !ruby/regexp /features\/.*.rb$/
+  - !ruby/regexp /test\/.*.rb$/
```
Comment (pathawks) [inline .rubocop.yml]: > Yep, you can safely exclude all of `test/` from `Metrics/ClassLength`.

You got it, boss :+1:

**Comment by parkr** [comment]
@pathawks ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ 

@jekyllbot: merge +dev

**Pull request merged**